js/data/bucket.js

```diff
@@ -128,19 +140,17 @@ Bucket.prototype.populateBuffers = function() {
 
 /**
  * Check if there is enough space available in the current array group for
- * `vertexLength` vertices. If not, append a new elementGroup. Should be called
- * by `populateBuffers` and its callees.
+ * `vertexLength` vertices. If not, append a new array group. Should be called
+ * by `populateArrays` and its callees.
  *
  * Array groups are added to this.arrayGroups[programName].
  * An individual array group looks like:
  * {
  *     index: number,
- *     layout: {
- *         layout: VertexArrayType,
- *         ?element: ElementArrayType,
- *         ?element2: ElementArrayType
- *     },
- *     paint: { [layerName]: PaintVertexArrayType, ...  }
+ *     vertexArray: VertexArrayType,
```

Comment:
Why did you choose to call this `vertexArray` instead of `layoutArray`? 

Conceptually the `paintArrays` and `layoutArray` are both vertex arrays. The distinguishing characteristic is that one contains data from paint properties and the other data from layout properties.